sys/kern/kern_intr.c

```diff
@@ -1521,8 +1522,10 @@ intr_event_handle(struct intr_event *ie, struct trapframe *frame)
 	td->td_intr_nesting_level--;
 
 	/* The interrupt is not aknowledged by any filter and has no ithread. */
-	if (!thread && !filter)
-		return (++ie->ie_stray);
+	if (!thread && !filter) {
+		++ie->ie_stray;
+		return (stray ? ie->ie_stray : 0);
+	}
```

Comment:
Kind of a wishlist item.  I'm looking at a device which is known to generate a stray interrupt during start.  I would rather like to have it counted, yet returning `FILTER_STRAY` causes all interrupt controllers to mask the interrupt.  Given the device, this would be Bad, as a result the present driver simply always returns `FILTER_HANDLED`.